### PR TITLE
fix: changed DB config

### PR DESCRIPTION
**Pull request by vj-codes**
#### Is your pull request related to a problem? Please describe
Fixed issue #27

**Steps done -**
- [x] Install psycopg2
- [x] Install flask-sqlalchemy
- [x] freeze requirements.txt
- [x] create models folder and add -> init.py, game_users.py, messages.py

**Steps remaining-**
- [ ] add tests

#### Checklist
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **[CONTRIBUTING](https://github.com/fbdevelopercircles/open-source-edu-bot/blob/master/CONTRIBUTING.md)** document.
- [ ] All new and existing tests passed.
- [ ] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by vj-codes** [comment]
@aboullaite went through CI, any idea why is the build failing didn't understand why exactly.
Also any resources to add tests?

**Comment by aboullaite** [comment]
The log says: `ERROR: Invalid requirement: 'Flask-SQLAlchemy>=2.4.4,2.4.5' (from line 16 of /requirements.txt)`

**Comment by aboullaite** [comment]
@vj-codes the build is still failing! is the app not complaining at your env ? do you have docker on your machine! You can probably try to run Dockerfile and add any missing dependencies.

File: requirements.txt [hunk 1]
```diff
@@ -13,4 +13,4 @@ Flask-Babel>=1.0.0,<1.1.0
 flake8>=3.8.3,<3.9.0
 python-dotenv>=0.14.0,<0.15.0
 psycopg2>=2.8.5,<2.8.6
-Flask-SQLAlchemy>=2.4.4,2.4.5
+Flask-SQLAlchemy>=2.4.4,<2.4.5
```
Comment (elinguiuriel) [inline requirements.txt:16]: Try
```
psycopg2>=2.8.5,<2.9.0
Flask-SQLAlchemy>=2.4.4,<2.5.0
```
Try it locally before pushing your code

File: requirements.txt [hunk 2]
```diff
@@ -11,4 +11,6 @@ pytest>=5.4.3,<5.5.0
 coverage>=5.1,<6.0
 Flask-Babel>=1.0.0,<1.1.0
 flake8>=3.8.3,<3.9.0
-python-dotenv>=0.14.0,<0.15.0
\ No newline at end of file
+python-dotenv>=0.14.0,<0.15.0
+psycopg2>=2.8.5,<2.8.6
```
Comment (elinguiuriel) [inline requirements.txt:15]: for pycopg2 to be installed you have to install these package
```bash
apt-get install postgresql-client gcc libc-dev linux-headers postgresql-dev 
```
or in alpine Dockerfile:

```
RUN apk add --update --no-cache postgresql-client
RUN apk add --update --no-cache --virtual .tmp-build-deps \
    gcc libc-dev linux-headers postgresql-dev 

```

**Comment by vj-codes** [comment]
@aboullaite @elinguiuriel I did not change anything, works completely fine locally.
I don't use docker , should the packages mentioned above be added ?
Screenshot (50)

**Comment by elinguiuriel** [comment]
> @aboullaite @elinguiuriel I did not change anything, works completely fine locally.
> I don't use docker , should the packages mentioned above be added ?
> Screenshot (50)

@vj-codes it's good to know that the code work locally, that does not guarantee that it will work on another computer. Maybe you use your computer for many other things and you already have the dependencies installed, think about those that do not have them already installed.
The code is tested and deployed using Docker to mimic installation in a totally new environment, if you pass the docker installation then we can be sure that the code will work merely everywhere. 

Just make sure to add the instructions described above in the 2 in `Dockerfile` and `dev.Dockerfile` and you may pass the tests

`dev.Dockerfile` 
```YAML
# Copyright (c) Facebook, Inc. and its affiliates.
#
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.

FROM python:3.7-alpine
LABEL MAINTAINER "Facebook Developers Circles" 

RUN apk add --update --no-cache postgresql-client
RUN apk add --update --no-cache --virtual .tmp-build-deps \
    gcc libc-dev linux-headers postgresql-dev 

RUN apk add --no-cache python3-dev libffi-dev gcc musl-dev make
RUN pip install gunicorn[gevent]

COPY requirements.txt /requirements.txt
RUN pip install -r /requirements.txt

WORKDIR /app
COPY ./src .

RUN pybabel compile -d locales

EXPOSE 5000

CMD gunicorn "fbosbot:create_app()" --worker-class gevent --workers 5 --bind 0.0.0.0:$PORT --max-requests 10000 --timeout 5 --keep-alive 5 --log-level info
```

`Dockerfile`
  
```YAML
# Copyright (c) Facebook, Inc. and its affiliates.
#
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.

FROM python:3.7-alpine
LABEL MAINTAINER "Facebook Developers Circles" 

WORKDIR /app

COPY ./src .

RUN apk add --update --no-cache postgresql-client
RUN apk add --update --no-cache --virtual .tmp-build-deps \
    gcc libc-dev linux-headers postgresql-dev 

COPY requirements.txt /requirements.txt
RUN pip install -r /requirements.txt
RUN pybabel compile -d locales
```

**Comment by vj-codes** [comment]
@elinguiuriel @aboullaite The build is successful now!
What shall be done for tests and I don't exactly understand what is to be done model structure mentioned in the issue.

**Comment by elinguiuriel** [comment]
> @elinguiuriel @aboullaite The build is successful now!
> What shall be done for tests and I don't exactly understand what is to be done model structure mentioned in the issue.

You only have to test like this when you install a new package. if possible uses docker to not be biased by your environment preinstalled dependencies 
When you simply modify code, make sure this command pass without error

```bash
(venv) $ pytest && flake8
```
the CI/CD [workflows](https://github.com/fbdevelopercircles/open-source-edu-bot/tree/9544c9a2b99234db6541cd929e6854d979a7868c/.github/workflows) use it to check PR and merge to the master branch

**Comment by vj-codes** [comment]
@elinguiuriel 5 tests passing one failing 
Screenshot (52)

**Comment by aboullaite** [comment]
@vj-codes What's the status of this one ? were you able to add @elinguiuriel changes ?

**Comment by vj-codes** [comment]
> @vj-codes What's the status of this one ? were you able to add @elinguiuriel changes ?

Yes @aboullaite I added it, everything is working good only concern is that one test is failing(shared screen shot above)

**Comment by Awinja-j** [comment]
@vj-codes  you need to set up verify_token in your env and It should work proper.

LGTM @elinguiuriel any more change requests?

**Pull request closed without merging**